### PR TITLE
[master] fix error "Self referencing loop detected with type"

### LINKED ISSUE #3532: GUI frontend on Linux looks broken
### Steps to Reproduce

1.Open otd-gui

2.Prompt "Application Error"
```
StreamJsonRpc.RemoteSerializationException: 写入 JSON RPC 消息时出错: JsonSerializationException: Self referencing loop detected with type 'OpenTabletDriver.Desktop.Interop.Display.Display'. Path ''.

   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at OpenTabletDriver.UX.App.Synchronize() in /build/opentabletdriver-git/src/OpenTabletDriver/OpenTabletDriver.UX/App.cs:line 192
   at OpenTabletDriver.UX.MainForm.OnConnected() in /build/opentabletdriver-git/src/OpenTabletDriver/OpenTabletDriver.UX/MainForm.cs:line 163
```

### Expected Behavior

Open the GUI frontend

### Observed Behavior

The driver seems to be working fine, but the GUI front end is broken
```
zncookie@Colorful-PC ~/Desktop> systemctl status --user opentabletdriver
* opentabletdriver.service - OpenTabletDriver Daemon
     Loaded: loaded (/usr/lib/systemd/user/opentabletdriver.service; enabled; preset: enabled)
     Active: active (running) since Tue 2024-11-19 18:40:53 CST; 20min ago
 Invocation: 060cae93eac64e8494d926749f6c0c27
   Main PID: 1418 (sh)
      Tasks: 20 (limit: 18649)
     Memory: 126M (peak: 136.5M)
        CPU: 2.688s
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/opentabletdriver.service
             |-1418 sh /usr/bin/otd-daemon
             `-1425 /usr/lib/opentabletdriver/OpenTabletDriver.Daemon

Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [LinuxArtistMode:Debug]        Input: [152x95@<76, 47.5>:0<C2><B0>]
Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [LinuxArtistMode:Debug]        Output: [1920x1080@<960, 540>]
Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [LinuxArtistMode:Debug]        LockAspectRatio: False
Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [LinuxArtistMode:Debug]        AreaClipping: True
Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [LinuxArtistMode:Debug]        AreaLimiting: True
Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [LinuxArtistMode:Debug]        LockToBounds: True
Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [Wacom CTL-472:Info]        Output mode: Artist Mode
Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [MouseBinding:Debug]        Button: Left
Nov 19 18:53:15 Colorful-PC otd-daemon[1425]: [Settings:Info]        Driver is enabled.
Nov 19 18:57:33 Colorful-PC otd-daemon[1425]: [IPC:Debug]        Connected to a client.
```

### Tablet Device

Wacom CTL-472

### Diagnostic Information

I can't do it, it prompts the same error

### Acknowledgements

- [X] I have installed OpenTabletDriver by following its [official installation instructions](https://opentabletdriver.net/).
- [X] I have checked the [OpenTabletDriver Wiki](https://opentabletdriver.net/Wiki) and respective FAQ pages, and my issue was not covered or actually fixed.
- [X] I have searched the existing issues and this new issue is not a duplicate of any.
- [X] I have written a concise and meaningful title.
- [X] I am on the latest version of OpenTabletDriver.

### PR DESCRIPTION
**Pull request by AkiSakurai**
see https://github.com/JamesNK/Newtonsoft.Json/issues/2432#issuecomment-1100433219 
fix #3532

### CODE REVIEW

**Review by InfinityGhost — DISMISSED** [review]

File: OpenTabletDriver.Desktop/Json/Converters/InterfaceConverter.cs [hunk 1]
```diff
@@ -5,9 +5,11 @@ namespace OpenTabletDriver.Desktop.Json.Converters
 {
     public class InterfaceConverter<TInterface, TClass> : JsonConverter<TInterface> where TInterface : class where TClass : TInterface, new()
     {
+        public override bool CanWrite => false;
+
         public override void WriteJson(JsonWriter writer, TInterface? value, JsonSerializer serializer)
         {
-            serializer.Serialize(writer, value);
+            throw new NotImplementedException();
```
[the remark this answers is not in the document]
Comment (gonX) [reply OpenTabletDriver.Desktop/Json/Converters/InterfaceConverter.cs:12]: According to the linked issue, this does seem correct. Using a custom Json _reader_ reads like you need to do `CanWrite = false` in which case `WriteJson(blabla)` should probably be NotImplemented. I would suggest adding a comment why we can't write, though.

[the remark this answers is not in the document]
Comment (X9VoiD) [reply OpenTabletDriver.Desktop/Json/Converters/InterfaceConverter.cs:12]: Specifying CanWrite here to false causes Json.NET to fallback to default serializer when writing. It shouldn't break.

**Review by gonX — DISMISSED** [review]
Approving given the above comments. I am of the understanding that this is correct and does not break anything.

I am not sure how to best test this, though.

File: OpenTabletDriver.Desktop/Json/Converters/InterfaceConverter.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (InfinityGhost) [reply OpenTabletDriver.Desktop/Json/Converters/InterfaceConverter.cs:12]: > Specifying CanWrite here to false causes Json.NET to fallback to default serializer when writing. It shouldn't break.

Was unaware of this behavior, would be nice to have this added as a comment

File: OpenTabletDriver.Desktop/Json/Converters/InterfaceConverter.cs [hunk 2]
```diff
@@ -5,9 +5,11 @@ namespace OpenTabletDriver.Desktop.Json.Converters
 {
     public class InterfaceConverter<TInterface, TClass> : JsonConverter<TInterface> where TInterface : class where TClass : TInterface, new()
     {
+        public override bool CanWrite => false;
```
Comment (gonX) [inline OpenTabletDriver.Desktop/Json/Converters/InterfaceConverter.cs:8]: to satisfy comment suggestion:
```suggestion
        // Newtonsoft.JSON custom `JsonConverter`s that only implement deserialization
        //      must specify CanWrite = false to use their default JSON writer
        // See upstream issue: https://github.com/JamesNK/Newtonsoft.Json/issues/2432
        public override bool CanWrite => false;
```

**Review by gonX — APPROVED** [review]
LGTM

**Pull request merged**